catalyst/dl/scripts/trace.py

```diff
@@ -35,23 +58,40 @@ def trace_model_from_checkpoint(
     experiment: Experiment = ExperimentType(config)
 
     print(f"Load model state from checkpoints/{checkpoint_name}.pth")
-    model = experiment.get_model(next(iter(experiment.stages)))
+    if stage is None:
+        stage = list(experiment.stages)[0]
+
+    model = experiment.get_model(stage)
     checkpoint = utils.load_checkpoint(checkpoint_path)
     utils.unpack_checkpoint(checkpoint, model=model)
 
+    runner: RunnerType = RunnerType()
+    runner.set_model_device(model, device)
+
+    if loader is None:
+        loader = 0
+    batch = experiment.get_native_batch(stage, loader)
+
     print("Tracing")
     traced = trace_model(
-        model, experiment, RunnerType,
+        model,
+        runner,
```

Comment:
do we need to pass `runner` for tracing? 
is it not enough to just pass `model` and `runner.predict_batch`? 
we can get device with `utils.get_device()`

Reply:
by this way we also will make our tracing support more general without any Experiment/Runner dependencies

Reply:
Yes, we do need a runner, because we trace not a Tensor, but catalyst dict with features and whatever.

> we can get device with utils.get_device()
no, because Runner may have `cuda` and we need to trace in `cpu`, or runner may have cpu, but fp16 needs `cuda`, etc


> by this way we also will make our tracing support more general without any Experiment/Runner dependencies

not agreed, because you may trace your model easily with torch.jit, but Runner is needed to make forward pass with dict with additional `any2device`

Reply:
so, do we need `runner` or we need just `runner.predict_batch`?
and do we need to carry about `runner.device` if we need to trace the model for prespecified one?

Reply:
Yes, we need predictions through the runner. Because you may implement it in other ways :)